hil-test/tests/get_time.rs

```diff
@@ -45,4 +53,37 @@ mod tests {
         assert!(t2 > t1);
         assert!((t2 - t1).to_millis() >= 500u64);
     }
+
+    #[cfg(systimer)]
+    #[test]
+    #[timeout(3)]
+    fn test_current_time_construct_systimer(ctx: Context) {
+        let t1 = esp_hal::time::current_time();
```

Comment:
What does this test test, actually? If it's only "we don't reset current_time", then the delay is redundant and all you want is "t2 >= t1". Other than that, we have plenty of "delay 500ms and assert that we really delayed for 500ms" tests elsewhere IMO and this just makes CI take half a second longer.

I would also not duplicate the test function, as only a single statement is different between them.

Reply:
500 millis is probably a bit long here but if we have a delay shouldn't we delay before we construct the timer?